velox/exec/tests/HashJoinTest.cpp

```diff
@@ -21,13 +21,68 @@
 #include "velox/exec/tests/utils/HiveConnectorTestBase.h"
 #include "velox/exec/tests/utils/PlanBuilder.h"
 #include "velox/expression/ExprToSubfieldFilter.h"
+#include "velox/functions/Udf.h"
+#include "velox/functions/prestosql/registration/RegistrationFunctions.h"
 
 using namespace facebook::velox;
 using namespace facebook::velox::exec;
 using namespace facebook::velox::exec::test;
 
 using facebook::velox::test::BatchMaker;
 
+namespace {
+
+template <typename T>
+struct AlwaysTrueSimpleFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  void call(
+      bool& out,
+      const arg_type<Generic<T1>>& /*lhs*/,
+      const arg_type<Generic<T1>>& /*rhs*/) {
+    // Custom simple function that will always return true.
+    out = true;
+  }
+};
+
+class AlwaysTrueVectorFunction : public exec::VectorFunction {
+ public:
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& /*outputType*/,
+      exec::EvalCtx* context,
+      VectorPtr* result) const override {
+    BaseVector::ensureWritable(rows, BOOLEAN(), context->pool(), result);
+    auto resultVector = (*result)->asUnchecked<FlatVector<bool>>();
+
+    rows.applyToSelected([&](auto row) { resultVector->set(row, true && !args[0]->isNullAt(row) && !args[1]->isNullAt(row)); });
```

Comment:
This is default null so true && !args[0]->isNullAt(row) && !args[1]->isNullAt(row)) is unnecessary , however I wasnt sure if null semantics were being honored so added that check. The test still fails if always true is returned.